### PR TITLE
UIREQ-1032, UIREQ-1034, UIREQ-1035 - Request detail view  update for Lending, Borrowing and Pickup DCB roles

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
Reshare DCB is a feature that allows sharing of assets within consortium.
Folio has primarily 3 roles in this feature - Lending, Borrowing, Pickup
**Lending Role** 

- Requested item would be a real item (i.e., an item from inventory).
- Requester would be a virtual patron (user type - DCB)

**Borrowing Role** 

- Requested item would be a virtual item
- Requester would be a real patron

**Pickup Role** 

- Requested item would be a virtual item (i.e., an item from inventory).
- Requester would be a virtual patron (user type - DCB)

More details of the feature are available at https://wiki.folio.org/display/FOLIJET/DCB+Integration

## Approach
1. Actions menu will display only "cancel request" menu item for all the roles explained above - hence the logic behind this implementation is a check if the request is virtual or item is virtual.
2. Title and item barcode are available as only text when in case of virtual item.
3. All virtual items have hardcoded "istanceId" and "holdingsRecordId".
4. All virtual patrons have their lastname hardcoded to "DcbSystem"

## Refs
[UIREQ-1032](https://issues.folio.org/browse/UIREQ-1032) -[DCB] Request details: Hide all actions except "Cancel Request" in Action menu (Lending library)
[UIREQ-1034](https://issues.folio.org/browse/UIREQ-1034) - [DCB] Request details: Updates for Action menu and DCB item links (Borrowing library)
[UIREQ-1035](https://issues.folio.org/browse/UIREQ-1035) - [DCB] Request details: Updates for Action menu and DCB item links (Pickup library)

## Screenshots
**Two-legged transaction**
**Lending flow -**

**Request Status** - Open-Not yet filled
Lending-library-Open-Not-Yet-filled

**Request Status** - Open-In transit
Lending-library-Open-In-transit

**Request Status** - Open-Awaiting pickup
Lending-library-Open-Awaiting-pickup

**Request Status** - Closed-Filled
Lending-library-Closed-Filled

**Borrowing & Pickup flow -**

**Request Status** - Open-Not yet filled
image

**Request Status** - Closed-Filled
image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/ViewRequest.test.js [hunk 1]
```diff
@@ -119,175 +134,529 @@ describe('ViewRequest', () => {
...
+          beforeEach(() => {
+            render(
+              <CommandList commands={defaultKeyboardShortcuts}>
+                <ViewRequest {...props} />
+              </CommandList>
+            );
           });
 
-          it('should not render "Duplicate" button', () => {
+          it('should not render "Dublicate" button', () => {
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply src/ViewRequest.test.js:222]: I see on the application that the button name is "Duplicate"

File: src/components/TitleInformation/TitleInformation.test.js [hunk 2]
```diff
@@ -196,6 +202,30 @@ describe('TitleInformation', () => {
...
+        />
+      );
+    });
+
+    it('should render "instanceId" not as a link', () => {
+      const expectedProps = {
+        value: title,
+      };
+
+      expect(KeyValue).toHaveBeenNthCalledWith(
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply src/components/TitleInformation/TitleInformation.test.js:221]: 1. The test case intends to check that it is a not a link.
2.  line#32 declares title as a mere string, against which the assertion is made.

I think this is sufficient.

**Review by BogdanDenis — APPROVED** [review]

File: src/ViewRequest.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply src/ViewRequest.test.js]: done.

File: src/ViewRequest.test.js [hunk 3]
```diff
@@ -119,175 +134,529 @@ describe('ViewRequest', () => {
...
+
+          it('should render action menu with only "Cancel request" button', () => {
+            expect(screen.getByText(labelIds.cancelRequest)).toBeInTheDocument();
+            expect(screen.queryByText(labelIds.edit)).not.toBeInTheDocument();
+            expect(screen.queryByText(labelIds.duplicateRequest)).not.toBeInTheDocument();
+            expect(screen.queryByText(labelIds.moveRequest)).not.toBeInTheDocument();
+            expect(screen.queryByText(labelIds.reorderQueue)).not.toBeInTheDocument();
+          });
+
+          // describe('when request is valid', () => {
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply src/ViewRequest.test.js]: Removed.

File: src/ViewRequest.test.js [hunk 4]
```diff
@@ -119,175 +134,529 @@ describe('ViewRequest', () => {
...
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply src/ViewRequest.test.js]: Removed.

**Review by mariia-aloshyna — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

**Pull request merged**